core/templates/dev/head/i18n.js

```diff
@@ -60,6 +61,26 @@ oppia.controller('I18nFooter', [
   });
 }]);
 
+// Service to store user preferred language when he/she is not logged in.
+// It is equal to angular-translate-storage-cookie, but ignores cookies
+// when the variable GLOBALS.IS_EMBEDDED is true
+oppia.factory('customCookieStorage', function($cookieStore) {
+  'use strict';
+  return {
+    put: function(name, value) {
+      $cookieStore.put(name, value);
+    },
+    get: function(name) {
+      // If the variable is not set assume true
+      if (GLOBALS.IS_EMBEDDED || GLOBALS.IS_EMBEDDED === undefined) {
```

Comment:
Yes I like that option much more! I completely missed this isIframed function, thanks for pointing it out!